datafusion/physical-plan/src/joins/hash_join/stream.rs

```diff
@@ -388,14 +430,20 @@ impl HashJoinStream {
                     handle_state!(ready!(self.fetch_probe_batch(cx)))
                 }
                 HashJoinStreamState::ProcessProbeBatch(_) => {
-                    let poll = handle_state!(self.process_probe_batch());
-                    self.join_metrics.baseline.record_poll(poll)
+                    handle_state!(self.process_probe_batch())
                 }
                 HashJoinStreamState::ExhaustedProbeSide => {
-                    let poll = handle_state!(self.process_unmatched_build_batch());
-                    self.join_metrics.baseline.record_poll(poll)
+                    handle_state!(self.process_unmatched_build_batch())
+                }
+                HashJoinStreamState::Completed => {
+                    // Flush any remaining buffered data
+                    if !self.output_buffer.is_empty() {
+                        self.output_buffer.finish_buffered_batch()?;
+                        // Continue loop to emit the flushed batch
+                        continue;
+                    }
+                    Poll::Ready(None)
                 }
```

Comment:
Cosmetic change, would change like this :

```suggestion
                HashJoinStreamState::Completed if !self.output_buffer.is_empty() => {
                    // Flush any remaining buffered data
                    self.output_buffer.finish_buffered_batch()?;
                    // Continue loop to emit the flushed batch
                    continue;
                }
                HashJoinStreamState::Completed => Poll::Ready(None),
```

make match slightly easier to read?  I guess it should not have performance implications. 

btw, moving `next_completed_batch()` to the top of the method makes it much easier to read and understand.